### PR TITLE
Incorporate `sc-peerset` into `sc-network`

### PR DESCRIPTION
**Pull request by dmitry-markin**
This is a preparation PR for https://github.com/paritytech/substrate/issues/14170. Nothing interesting is happening here, just `Peerset`, `PeerStore`, and `ProtocolController` are moved from `sc-peerset` to `sc-network` & `sc-network-common`.

Adding `B1-note_worthy` label because the `sc-peerset` crate is retired.

### CODE REVIEW

File: client/network/src/peerset.rs [hunk 1]
```diff
@@ -338,14 +312,6 @@ impl Peerset {
 		self.protocol_handles[set_id.0].dropped(peer_id);
 	}
 
-	/// Reports an adjustment to the reputation of the given peer.
-	pub fn report_peer(&mut self, peer_id: PeerId, score_diff: ReputationChange) {
```
Comment (altonen) [inline client/network/src/peerset.rs:342]: Why is this removed?

**Review by altonen — APPROVED** [review]

File: client/network/src/peerset.rs (hunk 1 above)
Comment (dmitry-markin) [reply client/network/src/peerset.rs:342]: All peer reporting is happening through the `PeersetHandle` now, so this function is not used. I don't know why the compiler didn't complain before.

File: client/network/src/protocol/notifications/behaviour.rs [hunk 2]
```diff
@@ -23,6 +23,7 @@ use crate::{
 	types::ProtocolName,
 };
 
+use crate::peerset::DropReason;
```
[the remark this answers is not in the document]
Comment (dmitry-markin) [reply client/network/src/protocol/notifications/behaviour.rs]: Done, thanks.

**Review by melekes — APPROVED** [review]
👍

**Comment by dmitry-markin** [comment]
IDK why, but after seemingly trivial changes the nodes in `reconnect_after_disconnect` `Notifications` test stopped receiving any network events. Investigating.

**Comment by altonen** [comment]
Maybe it was flakey/made flakey by some recent `sc-network` change and just got triggered in this PR?

**Comment by dmitry-markin** [comment]
> Maybe it was flakey/made flakey by some recent `sc-network` change and just got triggered in this PR?

It doesn't look like flaky, because it always succeeds in master, and always hangs in this PR.

File: client/network/src/peerset.rs [hunk 3]
```diff
@@ -292,9 +267,9 @@ impl Peerset {
 		let (protocol_handles, protocol_controllers): (Vec<ProtocolHandle>, Vec<_>) =
 			controllers.into_iter().unzip();
 
-		let (to_self, from_handle) = tracing_unbounded("mpsc_peerset_messages", 10_000);
+		let (tx, from_handle) = tracing_unbounded("mpsc_peerset_messages", 10_000);
 
-		let handle = PeersetHandle { tx: to_self.clone() };
+		let handle = PeersetHandle { tx };
```
Comment (altonen) [inline client/network/src/peerset.rs:272]: I think this change might be the issue, the print "`PeersetHandle` was dropped, terminating `Peerset` is being printed if you enable it.

Comment (dmitry-markin) [reply client/network/src/peerset.rs:272]: Thanks!!!
